src/components/Button/Button.tsx

```diff
@@ -20,10 +20,12 @@ interface BaseButtonProps {
   circle?: boolean;
   className?: string;
   darkMode?: boolean;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  forwardedRef?: React.Ref<any>;
```

Comment:
I need to go back to TS school. This component can spit out Refs to at least 3 element types, and I've had a helluva time getting them to play nicely when composing components.

If anyone feels really confident in this area, I'd welcome a helping hand.